torch/_inductor/codegen/cpp.py

```diff
@@ -4876,6 +4886,28 @@ def try_share_local_buffer(local_buffer_layout, local_buffers):
                     cpp_kernel_proxy_list.append(cpp_kernel_proxy)
                     nodes_list.append(_node.get_nodes())  # type: ignore[arg-type]
 
+                    outer_ranges = functools.reduce(
+                        lambda x, y: x * y,
+                        cpp_kernel_proxy.ranges[: node.outer_loop_fusion_depth],
+                    )
+                    # If the range of the first inner loop is much larger than
+                    # the range of all outer loops, fallback to standard codegen.
+                    if (
+                        len(cpp_kernel_proxy.ranges) > node.outer_loop_fusion_depth
+                        and isinstance(outer_ranges, sympy.Integer)
+                        and isinstance(
+                            cpp_kernel_proxy.ranges[node.outer_loop_fusion_depth],
+                            sympy.Integer,
+                        )
+                        and outer_ranges * 300
+                        < cpp_kernel_proxy.ranges[node.outer_loop_fusion_depth]
+                    ):
+                        for removed_buffer in scope.removed_buffers:
+                            # Restore the removed buffers by this context before
+                            # fallback to codegen without using Local Buffer
+                            V.graph.removed_buffers.remove(removed_buffer)
+                        return False
```

Comment:
Can we move it to another PR as it modify the outer loop fusion heuristic?

Reply:
Done.